### DISCUSSION TITLE
Not able to generate the PDF from HTML

### DISCUSSION THREAD

**Discussion by Jaimin-Devhero** [mpdf/mpdf#1672] [Q&A] [answered]
PHP: 8.0.19
"mpdf/mpdf": "^8.0",

I am generating the PDF from the HTML, but it's not generating for some reason. and generating the blank page in the response PDF.

**Below is my code:**

```
$html = '<table style="padding-right: 20px; padding-left: 20px; font-size:14px;font-family: helvetica">
    <tr>
        <td style="height:560px; ">
            <p>I hope you are getting well Reesha <3 You’ve been holding shit down since day1 gang I need you aliveeee buh hopefully you eating now too or imma be mad Luhhh youuuu. <<<3333 Adam Alexander Mlamali</p>
        </td>
    <tr>
</table>';

$mpdf->WriteFixedPosHTML($html, 105, 0, 104, 150, 'auto');
$mpdf->Output("", "I");
```

What I have figured out is that if I remove the "<<<" from the HTML its works fine. Can someone guide me what's the issue with <<< here and how I can generate the PDF without removing the "<<<"?

I have also tried to find the cause by enabling the debug for mPDf `$mpdf1->debug = true;` not found any issue exception in that also. (https://mpdf.github.io/troubleshooting/blank-screen.html)

**Comment by finwe** [comment, marked as answer]
Encode your HTML. Change all `<` characters that are not HTML tags to `&lt;`

**Reply by Jaimin-Devhero** [reply]
Hello @finwe 

What changes ii have to do in my code, can you please guide me in that?

**Reply by ssieb** [reply]
You need to convert those characters before you generate the html.  However you do it is very dependent on how you generate the html.

**Reply by Jaimin-Devhero** [reply]
hello 

My HTML is dynamically added from the CKEditor to the database. Then we are running a queue of jobs to generate PDFs for those HTML.

How can I convert those characters?

I have tried with htmlspecialchars(), htmlentities() but it's no use as it encodes the whole HTML.

**Reply by ssieb** [reply]
I can't tell you how to fix your system.  You should be storing the message content separately from the enclosing html.  If you aren't, then you have a big problem.

**Comment by Jaimin-Devhero** [comment]
Thanks, everyone for your help. It's solved.

Really appreciated.
